### PR TITLE
docs: update dev docs for protobuf changes

### PR DESCRIPTION
**Pull request by kruskall**
Update dev docs for protobuf changes

Closes https://github.com/elastic/apm-data/issues/73

### CODE REVIEW

File: dev_docs/HOW_TO.md [hunk 1]
```diff
@@ -11,14 +11,17 @@ Adding a new intake field requires changes to both apm-data and apm-server repos
    - RUM v3: [input/elasticapm/internal/modeldecoder/rumv3/model.go](/input/elasticapm/internal/modeldecoder/rumv3/model.go)
 2. Run `make generate` to generate the corresponding `model_generated.go` and JSON specification from the modified `model.go` in step 1.
 3. Run `make update-licenses` to add license header to `model_generated.go` generated in step 2.
-4. Add the new field to the corresponding file in `model/`, e.g. `span.go`.
-   1. Add the field to the struct.
-   2. Modify the `fields` function to include the field.
+4. Add the new field to the corresponding file in `model/proto/`, e.g. `span.proto`.
+   1. Run `make generate-modelpb` to generate the corresponding `span.pb.go`.
+   2. Run `make update-licenses fmt` to add license header to `span.pb.go` generated in step 1.
```
Comment (carsonip) [inline dev_docs/HOW_TO.md:16]: A little bit of ambiguity here, there's also a step 1 in the outer layer
```suggestion
   2. Run `make update-licenses fmt` to add license header to `span.pb.go` generated in the previous step.
```

File: dev_docs/HOW_TO.md [hunk 2]
```diff
@@ -75,9 +78,12 @@ Mapping an OTel field is similar to adding a field to Intake.
 ### 1. Make changes in apm-data (this repo)
 
 1. Modify OTel parsing code in [input/otlp](/input/otlp)
-2. Add the new field to the corresponding file in `model/`, e.g. `span.go`.
-   1. Add the field to the struct.
-   2. Modify the `fields` function to include the field.
+2. Add the new field to the corresponding file in `model/proto/`, e.g. `span.proto`.
+   1. Run `make generate-modelpb` to generate the corresponding `span.pb.go`.
+   2. Run `make update-licenses fmt` to add license header to `span.pb.go` generated in step 1.
```
Comment (carsonip) [inline dev_docs/HOW_TO.md:83]:
```suggestion
   2. Run `make update-licenses fmt` to add license header to `span.pb.go` generated in the previous step.
```

File: dev_docs/HOW_TO.md (hunk 1 above)
Comment (kruskall) [reply dev_docs/HOW_TO.md]: Thanks! Good catch :+1:

File: dev_docs/HOW_TO.md [hunk 3]
```diff
@@ -11,14 +11,17 @@ Adding a new intake field requires changes to both apm-data and apm-server repos
    - RUM v3: [input/elasticapm/internal/modeldecoder/rumv3/model.go](/input/elasticapm/internal/modeldecoder/rumv3/model.go)
 2. Run `make generate` to generate the corresponding `model_generated.go` and JSON specification from the modified `model.go` in step 1.
 3. Run `make update-licenses` to add license header to `model_generated.go` generated in step 2.
-4. Add the new field to the corresponding file in `model/`, e.g. `span.go`.
-   1. Add the field to the struct.
-   2. Modify the `fields` function to include the field.
+4. Add the new field to the corresponding file in `model/proto/`, e.g. `span.proto`.
+   1. Run `make generate-modelpb` to generate the corresponding `span.pb.go`.
```
Comment (kyungeunni) [inline dev_docs/HOW_TO.md:15]: Should we also mention that `span_vtproto.pb.go` would also be generated, and we prefer to use the methods from that file when needing to marshal and unmarshal with some brief introduction(as far as I know that was the case)?

Comment (kruskall) [reply dev_docs/HOW_TO.md:15]: > Should we also mention that span_vtproto.pb.go would also be generated,

Good catch! :+1: 

>  and we prefer to use the methods from that file when needing to marshal and unmarshal with some brief introduction(as far as I know that was the case)?

I planned to add that as comments/doc to the methods to make it more visible when using the library. Not sure if that approach makes sense, WDYT ?

Comment (simitt) [reply dev_docs/HOW_TO.md:15]: IMO this is worth both - adding comments and describing the difference and when to use what in the dev docs.

Comment (kyungeunni) [reply dev_docs/HOW_TO.md:15]: I'm with @simitt. As well as the comments on the methods, having an overview or a brief explanation about why we have `{entity}__vtproto.pb.go` files and why we choose to introduce them could help users to utilize them. (Based on my experience as I didn't know about this before Marc told me about it :) )

Comment (kruskall) [reply dev_docs/HOW_TO.md]: Sorry about that! I've added more sections :bow: 

Let me know what you think!

**Review by kyungeunni — APPROVED** [review]
LGTM!

**Pull request merged**